server/methods.js

```diff
@@ -19,46 +19,47 @@ Meteor.methods({
     return chatterUsers.length > 0;
   },
 
-  "message.build" (params) {
+  "message.send" (params) {
     check(params, {
       message: String,
       roomId: String
     });
 
-    const chatterId = getChatterId(Meteor.userId());
+    const {message, roomId} = params;
+    const chatterUserId = getChatterUserId(Meteor.userId());
 
-    if (!userInRoom(chatterId, params.roomId)) {
+    if (!userInRoom(chatterUserId, roomId)) {
       throw new Meteor.Error("user-not-in-room", "user must be in room to post messages");
     }
 
-    const message = new Chatter.Message({
-      message: params.message,
-      roomId: params.roomId,
-      userId: chatterId
+    const newMessage = new Chatter.Message({
+      userId: chatterUserId,
+      message,
+      roomId
     });
 
-    if (message.validate()) {
-      return message.save();
+    if (newMessage.validate()) {
```

Comment:
Is manual validation necessary? Doesn't Astronomy throw an error if the document is not valid? I'm not sure myself, but I think so.